backend/app/db/schemas/search.py

```diff
@@ -0,0 +1,115 @@
+from enum import Enum
+from typing import Dict, List, Optional, Tuple
+
+from pydantic import BaseModel
+
+
+class SortEnum(str, Enum):
+    """Sort ordering for use building OpenSearch query body."""
+
+    ASCENDING = "asc"
+    DESCENDING = "desc"
+
+
+class SortField(str, Enum):
+    """Sort field for use building OpenSearch query body."""
+
+    DATE = "date"
+    TITLE = "title"
+    TOP_HIT = "top_hit"
+    # TODO: complete enum
+
+
+class SearchRequestBody(BaseModel):
+    """The request body expected by the search API endpoint."""
+
+    query_string: str
+    exact_match: bool
+    max_passages_per_doc: int = 10  # TODO: decide on default
+
+    # TODO: Improve filters to allow generics & use filter types
+    keyword_filters: Optional[Dict[str, List[str]]] = None
+    year_range: Optional[Tuple[Optional[int], Optional[int]]] = None
+
+    sort_field: SortField = SortField.DATE
+    sort_order: SortEnum = SortEnum.DESCENDING
+
+    limit: int = 10  # TODO: decide on default
+    offset: int = 0
+
+
+class SearchResponseDocumentPassage(BaseModel):
+    """A Document passage match returned by the search API endpoint."""
+
+    text: str
+    text_block_id: str
+    text_block_page: int
```

Comment:
Note that the page number returned here is 0-indexed. This is maybe counter-intuitive: the front end would likely want to add 1 to this value for it to make sense to users. We should either add 1 in the API, or write in the API docs that it's 0-indexed?

Reply:
let's just add one in the processing - the backend code is already processing everything so it shouldn't make a massive difference

Reply:
by processing do you mean during indexing or during searching?

Reply:
SOrry, yes, I meant search time. When we process the result of the search from OpenSearch into the API response.